File: src/exercises/chapter1/lesson2.rb
# frozen_string_literal: true

# S(n) = 1^2 + 2^2 + … + n^2
# @param {Integer} n - The upper limit of the range
# @return {Integer} - The sum of squares of all integers from 1 to n

def solution(n)
  return 0 if n.zero?

  (1..n).sum { |i| i**2 }
end

File: src/exercises/chapter1/Lesson2.ts
/**
 * Lesson 2: Calculate S(n) = 1^2 + 2^2 + ... + n^2
 */
export default class Lesson2 {

  /**
   * This method calculates the sum of squares from 1 to n.
   *
   * @param {number} n - the number up to which squares are calculated.
   * @returns {number} the sum of squares from 1 to n.
   */
  solution(n: number): number {
    let result = 0

    // Loop through all numbers from 1 to n
    for (let i = 1; i <= n; i++) {
      // Add the square of the current number to the result
      result += i * i
    }

    // Return the final calculated result.
    return result
  }
}
